src/common/services/UtilService/UtilService.ts

```diff
@@ -5,10 +5,28 @@ import packageJson from '~/../../package.json';
 import { UnixTimestamp } from '~common/types/common';
 import { Seconds } from '~constants/Milliseconds';
 
+const PAUSE_IN_MILLISECONDS = 1;
 type NumberRange = [number, number];
 
 export class UtilService {
-  uuid = (): string => generateUUID.v1() as string;
+  uuid = (): string => {
+    try {
+      return generateUUID.v1() as string;
+    } catch (e: any) {
+      // catch the error "uuid.v1(): Can\'t create more than 10M uuids/sec"
+      if (e.message.includes('more than 10M uuids/sec')) {
+        // pause to allow the uuid to be generated and try again, throw error if it fails again
+        const start = new Date().getTime();
+        for (let i = 0; i < 1e7; i++) {
+          if (new Date().getTime() - start > PAUSE_IN_MILLISECONDS) {
+            break;
+          }
+        }
+        return generateUUID.v1() as string;
```

Comment:
oh gosh I should have looked at the implementation before using https://github.com/eugenehp/react-native-uuid/blob/36e31d95bd46d5d86f1a90a7e5d1d192fdb7e122/src/v1.ts#L93 :(